docs/features/anti-affinity/anti-affinity.md

```diff
@@ -0,0 +1,60 @@
+# Anti Affinity
+
+Depending on a cluster's configuration, a Blue Green Rollout (or a Canary rollout that uses traffic management) can cause newly created pods to restart after deploying a new version. This can be problematic, especially for applications that cannot startup quickly or do not gracefully exit.
+
+This behavior occurs because cluster auto-scaler wants to scale down the extra capacity which was created to support a Rollout
+running in double capacity. It usually happens if a Rollout has its own dedicated instance group since a Rollout has a greater effect on
+cluster auto-scaling. Therefore, clusters with a large pool of shared nodes experience the behavior less often.
+
+For example, here is a Rollout is running with 8 pods spread across 2 nodes. Each node can hold 6 pods:
+![ Original Rollout is running, spread across two nodes](images/step-0.png)
+
+When the `spec.template` of the Rollout changes, the controller creates a new ReplicaSet with the spec update and the total number of pods doubles. In this case, the number of pods increases
+to 16.
+
+Since each node can only hold 6 pods, the cluster autoscaler must increase the node count to 3 to accommodate
+all 16 pods. The resulting distribution of pods across nodes is shown here:
+![ Rollout running at double capacity](images/step-1.png)
+
+Once the Rollout finishes progressing, the old version is scaled down. This leaves the cluster with more nodes than necessary, thus wasting resources (as shown below).
+![ Original Rollout is running, spread across two nodes](images/step-2.png)
+
+The cluster auto-scaler terminates the extra node and the pods are rescheduled on the remaining 2 nodes.
+![ Original Rollout is running, spread across two nodes](images/step-3.png)
+
+To reduce the chance of this behavior, a rollout can inject anti-affinity. This prevents new pods on nodes which have the previous version's pods.
+
+You can learn more about anti-affinity [here](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/).
+
+Repeating the above example with anti-affinity enabled, here is what happens when the `.spec.template` of the Rollout changes. Due to anti-affinity, the new pods cannot be scheduled on nodes which run the old ReplicaSet's pods.
+As a result, cluster auto-scaler must create 2 nodes to host the new ReplicaSet's pods. 
```

Comment:
We should add one last comment saying that we wont restart the pods in this case because the nodes scaled down are guaranteed to not have the new Pods.